sdp.js

```diff
@@ -551,6 +551,55 @@ SDPUtils.parseMsid = function(mediaSection) {
   }
 };
 
+// SCTP
+// parses draft-ietf-mmusic-sctp-sdp-26 first and falls back
+// to draft-ietf-mmusic-sctp-sdp-05
+SDPUtils.parseSctpDescription = function(mediaSection) {
+  var mline = SDPUtils.parseMLine(mediaSection);
+  var maxSizeLine = SDPUtils.matchPrefix(mediaSection, 'a=max-message-size:');
+  var maxMessageSize;
+  if (maxSizeLine.length > 0) {
+    maxMessageSize = parseInt(maxSizeLine[0].substr(19), 10);
+  }
+  if (isNaN(maxMessageSize)) {
+    maxMessageSize = 65536;
+  }
+  var sctpPort = SDPUtils.matchPrefix(mediaSection, 'a=sctp-port:');
+  if (sctpPort.length > 0) {
+    return {
+      port: parseInt(sctpPort[0].substr(12), 10),
+      protocol: mline.fmt,
+      maxMessageSize: maxMessageSize
+    };
+  }
+  var sctpMapLines = SDPUtils.matchPrefix(mediaSection, 'a=sctpmap:');
+  if (sctpMapLines.length > 0) {
+    var parts = SDPUtils.matchPrefix(mediaSection, 'a=sctpmap:')[0]
+      .substr(10)
+      .split(' ');
+    return {
+      port: parseInt(parts[0], 10),
+      protocol: parts[1],
+      maxMessageSize: maxMessageSize
+    };
+  }
+};
+
+// SCTP
+// outputs the draft-ietf-mmusic-sctp-sdp-26 version that all browsers
```

Comment:
I have another question on this one... when parsing the old format, could the following happen:
- browser receives old format from old server
- browser sends answer in new format

Reply:
It does look like Chrome and Firefox will still answer with the old format if that was used in the offer. So this change would silently remap it to the new format the old server can't handle. :/

Add an original format flag in the parsed sctp object?

Reply:
is there a property in the old format that isn't in the new format?

Reply:
There was the streams value that got removed during review. Otherwise the same. Unless you want to interpret media protocol of DTLS/SCTP as old format?

Reply:
that sounds like the best way to handle this

Reply:
Good catch @fippo. Yes, we could hand out and parse a `legacy` flag. A boolean might be risky though... you never know if they'll change it again... :wink:

Reply:
Does this mean I would need to parse and save the sctp streams value again to preserve for outputting this case? Or is it fine to drop in an answer?

Reply:
Nah, don't bother about parsing and just write `65535` (precisely this one, not `65536`) in the legacy format. I don't think anyone parses that since it's irrelevant after the SCTP handshake.